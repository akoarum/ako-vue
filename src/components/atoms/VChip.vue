<template>
  <i :class="styledClasses" class="vChip"><slot /></i>
</template>

<script lang="ts">
import Vue from 'vue'

export default Vue.extend({
  props: {
    primary: { type: Boolean },
    secondary: { type: Boolean },
    caution: { type: Boolean }
  },
  computed: {
    styledClasses(): string[] {
      const classes: string[] = []

      if (this.primary) classes.push('-primary')
      if (this.secondary) classes.push('-secondary')
      if (this.caution) classes.push('-caution')

      return classes
    }
  }
})
</script>

<style lang="scss" scoped>
.vChip {
  display: inline-block;
  border-radius: 2px;
  background: #f0f0f0;
  line-height: 1;

  @include media-min {
    padding: 6px 10px 7px;
    font-size: 1.4rem;
  }
}

.-primary {
  background: lighten($color-theme, 44%);
}

.-secondary {
  background: lighten($color-secondary, 58%);
}

.-caution {
  background: lighten($color-caution, 52%);
}
</style>
