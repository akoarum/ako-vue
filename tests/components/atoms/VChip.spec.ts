import { shallowMount } from '@vue/test-utils'
import VChip from '~/components/atoms/VChip.vue'

describe('VChip', () => {
  let wrapper: any

  beforeEach(() => {
    wrapper = shallowMount(VChip, {
      propsData: {},
      slots: {
        default: 'テスト'
      }
    })
  })

  describe('initialized', () => {
    it('mountable', () => {
      expect(wrapper.isVueInstance()).toBe(true)
    })

    it('snapshot', () => {
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('computed', () => {
    it('styledClasses: propsを基にクラス配列を返す', () => {
      expect(wrapper.vm.styledClasses).toEqual([])

      wrapper.setProps({ primary: true })
      expect(wrapper.vm.styledClasses).toEqual(['-primary'])

      wrapper.setProps({ primary: false, secondary: true })
      expect(wrapper.vm.styledClasses).toEqual(['-secondary'])

      wrapper.setProps({ secondary: false, caution: true })
      expect(wrapper.vm.styledClasses).toEqual(['-caution'])
    })
  })
})
