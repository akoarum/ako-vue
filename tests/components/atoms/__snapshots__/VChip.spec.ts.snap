// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VChip initialized snapshot 1`] = `
<i
  class="vChip"
>
  テスト
</i>
`;
